### PR TITLE
Custom repositories for prcheck and compatibility tests

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
Adds functionality for prcheck and copmpatibility tests to be able to deploy custom registries alongside the rh-che deployment

### What issues does this PR fix or reference?
#1658 

### How have you tested this PR?
manual testing

### CODE REVIEW

**Comment by Katka92** [comment]
[test]

**Comment by ScrewTSW** [comment]
[test]

**Comment by ScrewTSW** [comment]
[test]

File: dev-scripts/deploy_custom_rh-che.sh [hunk 1]
```diff
@@ -326,6 +396,12 @@ CHE_CONFIG_YAML=$(echo "$CHE_CONFIG_YAML" | \
                       .\"metadata\".\"name\" = \"rhche\" |
                       .\"data\".\"CHE_INFRA_OPENSHIFT_TLS__ENABLED\" = \"$RH_CHE_USE_TLS\" ")
 
+if [ "$RH_CHE_USE_CUSTOM_REPOSITORIES" == "true" ]; then
```
Comment (ibuziuk) [inline dev-scripts/deploy_custom_rh-che.sh]: REPOSITORIES? maybe registries

File: dev-scripts/deploy_custom_rh-che.sh [hunk 2]
```diff
@@ -74,11 +80,16 @@ function unsetVars() {
   unset RH_CHE_USE_TLS;
   unset RH_CHE_APPLY_SECRET;
   unset RH_CHE_OC_SECRET;
+  unset RH_CHE_USE_CUSTOM_REPOSITORIES;
+  unset RH_CHE_VERSION;
+  unset RH_CHE_CUSTOM_REPOSITORIES_VERSION;
```
Comment (ibuziuk) [inline dev-scripts/deploy_custom_rh-che.sh]: REPOSITORIES_VERSION?

File: dev-scripts/deploy_custom_rh-che.sh [hunk 3]
```diff
@@ -280,6 +304,52 @@ fi
 
 echo -e "\\033[92;1mGetting deployment scripts done.\\033[0m"
 
+# DEPLOY CUSTOM REPOSITORIES
```
Comment (ibuziuk) [inline dev-scripts/deploy_custom_rh-che.sh]: REGISTRIES ?

**Comment by ibuziuk** [comment]
@ScrewTSW :+1: but the `RH_CHE_USE_CUSTOM_REPOSITORIES` name is misleading
please take a look at comments

File: dev-scripts/deploy_custom_rh-che.sh (hunk 1 above)
Comment (ScrewTSW) [reply dev-scripts/deploy_custom_rh-che.sh]: fixed

File: dev-scripts/deploy_custom_rh-che.sh (hunk 2 above)
Comment (ScrewTSW) [reply dev-scripts/deploy_custom_rh-che.sh]: fixed

File: dev-scripts/deploy_custom_rh-che.sh [hunk 4]
```diff
@@ -74,11 +80,16 @@ function unsetVars() {
   unset RH_CHE_USE_TLS;
   unset RH_CHE_APPLY_SECRET;
   unset RH_CHE_OC_SECRET;
+  unset RH_CHE_USE_CUSTOM_REPOSITORIES;
```
[the remark this answers is not in the document]
Comment (ScrewTSW) [reply dev-scripts/deploy_custom_rh-che.sh]: fixed

File: dev-scripts/deploy_custom_rh-che.sh (hunk 3 above)
Comment (ScrewTSW) [reply dev-scripts/deploy_custom_rh-che.sh]: fixed

**Comment by ScrewTSW** [comment]
> @ScrewTSW +1 but the `RH_CHE_USE_CUSTOM_REPOSITORIES` name is misleading
> please take a look at comments

Yeah, sorry about that ^^;;

**Pull request merged**